includes/init_includes/init_special_funcs.php

```diff
@@ -7,56 +7,66 @@
  * @copyright Copyright 2003-2012 Zen Cart Development Team
  * @copyright Portions Copyright 2003 osCommerce
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version GIT: $Id: Author: Ian Wilson  Tue Jul 24 11:36:47 2012 +0100 Modified in v1.5.1 $
+ * @version GIT: $Id: Author: Ian Wilson  Tue Jul 24 11:36:47 2012 +0100 Modified in v1.5.1 $
  */
 if (!defined('IS_ADMIN_FLAG')) {
-  die('Illegal Access');
+    die('Illegal Access');
 }
 /**
  * require the whos online functions and update
  */
-require(DIR_WS_FUNCTIONS . 'whos_online.php');
+require DIR_WS_FUNCTIONS . 'whos_online.php';
 zen_update_whos_online();
 /**
  * require the banner functions, auto-activate and auto-expire
  */
-require(DIR_WS_FUNCTIONS . 'banner.php');
+require DIR_WS_FUNCTIONS . 'banner.php';
 zen_activate_banners();
 zen_expire_banners();
+/**
+ * require product functions one time such that if previously loaded will not cause an error here.
+ */
+require_once DIR_WS_FUNCTIONS . 'functions_products.php';

```

Comment:
@mc12345678 
Given that `init_general_funcs.php` already has a `require` for `functions_products.php` which loads at breakpoint `60`, why are we requiring it again here again in `init_speical_funcs.php` which loads at breakpoint `150`?

Ref: https://github.com/zencart/zencart/pull/2551

Reply:
In essence, to allow such a comment/question/tracking to identify the dependency of these changes on `functions_products.php`... Possibly could have just added a note to indicate that dependency; however, also wanted to be sure that it was actually included at or before the subsequent code in the event the file's load point was otherwise changed.  I recall there being some "adjustments" being made to its (`functions_products.php`) load point and/or inclusion.

As a "final product" I would say that identifying the dependency as a code comment should be sufficient and that the line could otherwise be removed.